osvimdriver/openstack/neutron/driver.py

```diff
@@ -19,23 +22,46 @@ def get_network_by_id(self, network_id):
             raise ValueError('network_id must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving network with id %s', network_id)
-        result = neutron_client.show_network(network_id)
-        return result['network']
+        try:
+            external_request_id = str(uuid.uuid4())
+            driver_request_id  = str(uuid.uuid4())
+            self._generate_additional_logs('', 'sent', external_request_id, '',
+                                       'request', 'http', {'method' : 'get', 'uri' : LOG_URI_PREFIX +'/networks/' + network_id }, driver_request_id)
+            result = neutron_client.show_network(network_id)
+            self._generate_additional_logs(result, 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : 200, 'status_reason_phrase' : 'ok'}, driver_request_id)  
+            return result['network']
+        except Exception as e:
+            self._generate_additional_logs(e, 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : e.status_code,'status_reason_phrase' : e.message }, driver_request_id)
+            raise e
+
 
     def get_network_by_name(self, network_name):
         if network_name is None:
             raise ValueError('network_name must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving network with name %s', network_name)
-        result = neutron_client.list_networks()
+        external_request_id = str(uuid.uuid4())
+        driver_request_id  = str(uuid.uuid4())
```

Comment:
driver_request_id is specific to a lifecycle, Please pass this value from the calling function.

Reply:
I have done changes.